File: php/sort/src/SelectionSort/SelectionSort.php
<?php

namespace Sort\SelectionSort;

class SelectionSort
{
    /**
     * 插入排序。
     *
     * @param array $array 待排序数组
     *
     * @return array
     */
    public static function sort(array $array): array
    {
        if (count($array) < 2) {
            return $array;
        }

        $result = [];
        foreach ($array as $item) {
            $index = self::findSmallestIndex($array);
            $result[] = array_splice($array, $index, 1)[0];
        }

        return $result;
    }

    /**
     * 查找最小值的索引。
     *
     * @param array $array 待排序的数组
     *
     * @return int 返回数组中最小元素的索引
     */
    public static function findSmallestIndex(array & $array): int
    {
        [$smallest, $index] = [$array[0], 0];
        foreach ($array as $key => $value) {
            if ($value < $smallest) {
                $smallest = $value;
                $index = $key;
            }
        }

        return $index;
    }
}

File: python/sort/b/selection_sort.py
"""
选择排序
"""


def selection_sort(items: list) -> list:
    for i in range(len(items) - 1, 0, -1):
        k = 0
        for j in range(1, i + 1):
            if items[j] > items[k]:
                k = j
        items[i], items[k] = items[k], items[i]
    return items


def selection_sort2(items: list) -> list:
    result = []
    for i in range(len(items)):
        smallest = find_smallest(items)
        result.append(items.pop(smallest))
    return result


def find_smallest(items: list) -> int:
    index, smallest = 0, items[0]
    for i in range(1, len(items)):
        if items[i] < smallest:
            index, smallest = i, items[i]
    return index
